### DISCUSSION TITLE
Retrieve the redirect history?

### DISCUSSION THREAD

**Discussion by aw-was-here** [encode/httpx#2696] [Ideas]
Hello!

I'm looking at how `httpx` handles redirects and there are some custom things I'd like to do, such as letting scheme up/down grades be "free."  Since hooks are fired off after every redirect (hooray!),  it is certain possible to do my own record keeping and counting... but it would seem to be a lot easier to just have the redirection history exposed from inside the request that is passed to `hook`.  Is there any downside to doing so? Would a PR be accepted?

Thanks!

**Comment by lovelydinosaur** [comment]
> have the redirection history exposed from inside the request

We do expose the redirect history on the `Response` instance, with `response.history`...

https://www.python-httpx.org/quickstart/#redirection-and-history

I'm not sure if that's what you're looking for or not.
